### ISSUE TITLE
Add `isLoading` and `stop` functionality to `useObject` hook

### ISSUE THREAD

**Issue by coji** [vercel/ai#2082]
### Feature Description

I propose adding two new features to the `useObject` hook in the Vercel AI SDK:

1. `isLoading`: A boolean state that indicates whether the object is currently being generated or updated.
2. `stop`: A function that allows users to interrupt the ongoing object generation or update process.

These additions would align the `useObject` hook more closely with other hooks in the SDK, such as `useChat` and `useCompletion`, which already have similar functionalities.

### Use Case

Adding `isLoading` and `stop` to `useObject` would significantly improve the user experience in applications using this hook. Here are some specific use cases:

1. **Loading State UI**: With `isLoading`, developers can easily implement loading indicators or disable UI elements while the object is being generated or updated. This provides clear feedback to users about the current state of the operation.

   ```tsx
   {isLoading ? <LoadingSpinner /> : <GenerateButton onClick={handleGenerate} />}
   ```

2. **Cancellation**: The `stop` function would allow users to cancel long-running or unnecessary object generations, giving them more control over the application's behavior and potentially saving computational resources.

   ```tsx
   <button onClick={stop} disabled={!isLoading}>Cancel Generation</button>
   ```

3. **Error Handling**: `isLoading` can be used in conjunction with error states to provide more nuanced feedback to users, distinguishing between "loading", "error", and "ready" states.

   ```tsx
   {isLoading ? <LoadingSpinner /> : error ? <ErrorMessage /> : <ObjectDisplay />}
   ```

These enhancements would make the `useObject` hook more versatile and user-friendly, aligning it with the high standards of other hooks in the Vercel AI SDK. I believe this addition would be valuable for many developers using the Vercel AI SDK in their projects.

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2090

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2098

**Comment by coji** [comment]
@lgrammel Thanks for the lightning-fast implementation! Really appreciate it.​​​​​​​​​​​​​​​​

**Comment by mirhadiseyidli** [comment]
@lgrammel Hi. Sorry for bringing up an old thread, I'm having slight issue with stop function. I use useObject and streamText (express backend) and stop() call doesn't seem to send the abort control. The frontend stops the streaming but backend keeps going until the end

**Comment by lgrammel** [comment]
@mirhadiseyidli most likely a backend issue, you need to look into how forwarding fetch signals works with express and then pass an abort signal into streamText in the backend

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Feature**  
Confidence: **High**

### Rationale
This is a request to expand the `useObject` API with loading-state and cancellation capabilities, rather than a report that existing promised behavior is broken.

### Evidence
- The issue explicitly proposes adding new `isLoading` state and `stop` functionality to `useObject` for parity with `useChat` and `useCompletion`.
- The requested capabilities were subsequently addressed through linked pull requests #2090 and #2098.
- A later cancellation concern was attributed by a maintainer to Express abort-signal forwarding and backend configuration, not a confirmed SDK defect.
